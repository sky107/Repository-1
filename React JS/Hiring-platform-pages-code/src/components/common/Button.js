import React from 'react';
import classes from './Button.module.css';
const Button=(props)=>{
return(<>
<button {...props} className={classes[`${props.type}`]}>{props.children}</button>
</>);


}
export default Button;