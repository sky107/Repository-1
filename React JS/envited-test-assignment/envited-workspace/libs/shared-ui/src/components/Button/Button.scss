.btn {
    
    font-family: Helvetica, sans-serif;
    padding: 16px 48px;
    margin-top: 32px;
    outline: none;
    border: none;
    // display: flex;
    // flex-direction: row;
    justify-content: center;
    align-items: center;
    // padding: 16px;
    gap: 10px;
    font-size: 20px;
    line-height: 23px;
    // position: absolute;
    // width: 320px;
    // height: 55px;
    // left: calc(50% - 320px/2 + 436px);
    // top: 622px;
    background: linear-gradient(90deg, #8456EC 3.25%, #E87BF8 100%);
    border-radius: 10px;
    color: white;
    cursor: pointer;
}