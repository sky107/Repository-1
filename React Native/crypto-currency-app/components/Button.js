import { ThemeProvider } from '@react-navigation/native';
import React from 'react';
import { View,StyleSheet } from 'react-native';
import { TouchableOpacity } from 'react-native';
import {theme} from '../constants';
import { Colors } from '../constants/colors';
import Text from './Text';

export default function Button({
    children,
    color,
    textColor,
    textCenter,
    bold,
    style,
    ...props
}) {
  return (
    <TouchableOpacity
    style={[styles.buttonStyles,style]}
    >
       <Text color={textColor} textCenter={textCenter}>{children}</Text>
    </TouchableOpacity>
  );
}


const styles=StyleSheet.create({
buttonStyles:{
    backgroundColor:Colors.primary,
    padding:10,
    margin:10

}
})
