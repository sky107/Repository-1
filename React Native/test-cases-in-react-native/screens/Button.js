import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
function AppButton({ onPress }) {

  const printName = () => {
    return "SIddharth"
  }
  return (
    <TouchableOpacity
      style={[styles.button,
      {}]}
      onPress={onPress} >
      <Text style={styles.text}>Register</Text>
    </TouchableOpacity>
  );
}
const styles = StyleSheet.create({
  button: {
    backgroundColor: 'dodgerblue',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: '#fff',
    padding: 10
  }
})
export default AppButton;
