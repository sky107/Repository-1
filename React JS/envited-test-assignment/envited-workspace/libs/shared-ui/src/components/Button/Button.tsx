import React from 'react';
import './Button.scss'

interface ButtonProps{
    children?:string;
    onClick?:any;
}



export const Button=(props:ButtonProps)=>{

    const {onClick}=props;

    return <button className="btn" onClick={()=>onClick()}>
            {props.children}
        </button>
}