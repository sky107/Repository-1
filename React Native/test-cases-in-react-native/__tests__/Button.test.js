import React from "react";
import renderer from "react-test-renderer";
import Button from '../screens/Button';
it('renders correctly across screens', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});