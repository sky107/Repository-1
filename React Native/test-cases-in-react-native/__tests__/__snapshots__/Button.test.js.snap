// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "dodgerblue",
      "borderRadius": 12,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "padding": 10,
      }
    }
  >
    Register
  </Text>
</View>
`;
